test/distributed/tensor/parallel/test_micro_pipeline_tp.py

```diff
@@ -56,6 +56,15 @@ def _fp8_all_gather(tensor: torch.Tensor, gather_dim: int, group_name: str):
     return torch.cat(chunks, dim=gather_dim).view(tensor.dtype)
 
 
+def is_rocm_testing_and_without_new_arch():
```

Comment:
@BLOrange-AMD I have a few different issues with this function:
1. This should really be a decorator like `skipIfRocmArch` 
2. This seems like generic-enough logic which belongs in `torch/tools/testing/_internal`
3. Instead of hardcoding the pattern, it might make sense to have the decorator accept a list of strings to use to match

Reply:
@jithunnair-amd PR has been revised accordingly.